numpyro/contrib/control_flow/util.py

```diff
@@ -0,0 +1,45 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from jax.tree_util import register_pytree_node_class
+
+
+@register_pytree_node_class
+class PytreeTrace:
```

Comment:
Since this was no longer specific to `scan` I moved it into a separate file, what do you think?

Reply:
Sure, please go ahead.